TidepoolKit/Internal/Extensions/Date.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  Date.swift
+//  TidepoolKit
+//
+//  Created by Darin Krauss on 11/5/19.
+//  Copyright © 2019 Tidepool Project. All rights reserved.
+//
+
+import Foundation
+
+public extension Date {
+    init?(timeString: String) {
+        if let date = Date.timeFormatter.date(from: timeString) {
+            self = date.roundedToTimeInterval(TimeInterval.millisecond)
+        } else if let date = Date.timeFormatterAlternate.date(from: timeString) {
+            self = date.roundedToTimeInterval(TimeInterval.millisecond)
+        } else {
+            return nil
+        }
+    }
+
+    init?(deviceTimeString: String) {
+        if let date = Date.deviceTimeFormatter.date(from: deviceTimeString) {
+            self = date.roundedToTimeInterval(TimeInterval.millisecond)
+        } else if let date = Date.deviceTimeFormatterAlternate.date(from: deviceTimeString) {
+            self = date.roundedToTimeInterval(TimeInterval.millisecond)
+        } else {
+            return nil
+        }
+    }
+
+    var timeString: String {
+        return Date.timeFormatter.string(from: self.roundedToTimeInterval(TimeInterval.millisecond))
+    }
+
+    var deviceTimeString: String {
+        return Date.deviceTimeFormatter.string(from: self.roundedToTimeInterval(TimeInterval.millisecond))
+    }
+
+    static let timeFormatter: ISO8601DateFormatter = {
```

Comment:
These two Date formats cover all of the Tidepool data model and are not version dependent. However, I just noticed the extension is `public` and need not be. I will change to avoid cluttering the public namespace.